src/MaterialDesignControls/ControlsMaterial3/BaseMaterialFieldControl.xaml.cs

```diff
@@ -0,0 +1,687 @@
+        {
+            get { return !string.IsNullOrEmpty(LeadingIcon) || CustomLeadingIcon != null; }
+        }
+
+        public static readonly BindableProperty LeadingIconCommandProperty =
+            BindableProperty.Create(nameof(LeadingIconCommand), typeof(ICommand), typeof(BaseMaterialFieldControl));
+
+        public ICommand LeadingIconCommand
+        {
+            get { return (ICommand)GetValue(LeadingIconCommandProperty); }
+            set { SetValue(LeadingIconCommandProperty, value); }
+        }
+
+        public static readonly BindableProperty LeadingIconCommandParameterProperty =
+            BindableProperty.Create(nameof(LeadingIconCommandParameter), typeof(object), typeof(BaseMaterialFieldControl), defaultValue: null);
+
+        public object LeadingIconCommandParameter
+        {
+            get { return GetValue(LeadingIconCommandParameterProperty); }
+            set { SetValue(LeadingIconCommandParameterProperty, value); }
+        }
+
+        public static readonly BindableProperty TrailingIconProperty =
+            BindableProperty.Create(nameof(TrailingIcon), typeof(string), typeof(BaseMaterialFieldControl), defaultValue: null);
+
+        public string TrailingIcon
+        {
+            get { return (string)GetValue(TrailingIconProperty); }
+            set { SetValue(TrailingIconProperty, value); }
+        }
+
+        public static readonly BindableProperty CustomTrailingIconProperty =
+            BindableProperty.Create(nameof(CustomTrailingIcon), typeof(View), typeof(BaseMaterialFieldControl), defaultValue: null);
+
+        public View CustomTrailingIcon
+        {
+            get { return (View)GetValue(CustomTrailingIconProperty); }
+            set { SetValue(CustomTrailingIconProperty, value); }
+        }
+
+        public bool TrailingIconIsVisible
+        {
+            get { return !string.IsNullOrEmpty(TrailingIcon) || CustomTrailingIcon != null; }
+        }
+
+        public static readonly BindableProperty TrailingIconCommandProperty =
+            BindableProperty.Create(nameof(TrailingIconCommand), typeof(ICommand), typeof(BaseMaterialFieldControl));
+
+        public ICommand TrailingIconCommand
+        {
+            get { return (ICommand)GetValue(TrailingIconCommandProperty); }
+            set { SetValue(TrailingIconCommandProperty, value); }
+        }
+
+        public static readonly BindableProperty TrailingIconCommandParameterProperty =
+            BindableProperty.Create(nameof(TrailingIconCommandParameter), typeof(object), typeof(BaseMaterialFieldControl), defaultValue: null);
+
+        public object TrailingIconCommandParameter
+        {
+            get { return GetValue(TrailingIconCommandParameterProperty); }
+            set { SetValue(TrailingIconCommandParameterProperty, value); }
+        }
+
+        #endregion Icons
+
+        #region LabelPlaceholderText
+        public static readonly BindableProperty LabelPlaceholderTextProperty =
```

Comment:
Can we use only the PlaceholderText and add a new boolean property AnimatePlaceholder in order to handle this?